toolkit/tools/pkg/imagecustomizerlib/downloadimageazurelinux.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/microsoft/azure-linux-image-tools/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azure-linux-image-tools/toolkit/tools/internal/resources"
+)
+
+func downloadAzureLinuxImage(ctx context.Context, inputImage imagecustomizerapi.AzureLinuxImage, buildDir string,
+	imageCacheDir string,
+) (string, error) {
+	ociImage, err := generateAzureLinuxOciUri(inputImage)
+	if err != nil {
+		return "", err
+	}
+
+	signatureOptions := &ociSignatureCheckOptions{
+		TrustPolicyName:   "mcr-azure-linux",
+		TrustStoreName:    "microsoft-supplychain",
+		CertificateFs:     resources.ResourcesFS,
+		CertificateFsPath: resources.MicrosoftSupplyChainRSARootCA2022File,
+	}
+
+	inputImageFilePath, err := downloadOciImage(ctx, ociImage, buildDir, imageCacheDir, signatureOptions)
+	if err != nil {
+		return "", err
+	}
+
+	return inputImageFilePath, err
+}
+
+func generateAzureLinuxOciUri(inputImage imagecustomizerapi.AzureLinuxImage) (imagecustomizerapi.OciImage, error) {
+	majorMinor, date, err := inputImage.ParseVersion()
+	if err != nil {
+		return imagecustomizerapi.OciImage{}, err
+	}
+
+	tag := "latest"
+	if date != "" {
+		tag = majorMinor + "." + date
+	}
+
+	// Note: 'majorMinor' and 'tag' are already sanitized. So, there is no need to escape the values.
+	uri := fmt.Sprintf("mcr.microsoft.com/azurelinux/%s/image/minimal-os:%s", majorMinor, tag)
```

Comment:
So this will be dedicated to `minimal-os`? Should it be generic usage for variant that user supplied?

```
uri := fmt.Sprintf("mcr.microsoft.com/azurelinux/%s/image/%s:%s", majorMinor, inputImage.Variant, tag)
```

Reply:
Thanks! Fixed